pulse2percept/stimuli/base.py

```diff
@@ -0,0 +1,465 @@
+        Stimulus(compressed=False, data=[[...3.45]], electrodes=[0],
+                 interp_method='linear', shape=(1, 1), time=[...3.45])
+
+        Use the 'nearest' interpolation method instead
+        (see `scipy.interpolate.interp1d <https://docs.scipy.org/doc/scipy/reference/generated/scipy.interpolate.interp1d.html>`_):
+
+        >>> from pulse2percept.stimuli import Stimulus
+        >>> stim = Stimulus(np.arange(10).reshape((1, -1)),
+        ...                 interp_method='nearest')
+        >>> stim.interp(time=3.45) # doctest: +NORMALIZE_WHITESPACE +ELLIPSIS
+        Stimulus(compressed=False, data=[[...3.]], electrodes=[0],
+                 interp_method='linear', shape=(1, 1), time=[...3.45])
+
+        Extrapolate to a time point outside the provided data range:
+
+        >>> from pulse2percept.stimuli import Stimulus
+        >>> stim = Stimulus(np.arange(10).reshape((1, -1)))
+        >>> stim.interp(time=123.45) # doctest: +NORMALIZE_WHITESPACE +ELLIPSIS
+        Stimulus(compressed=False, data=[[...123.45]], electrodes=[0],
+                 interp_method='linear', shape=(1, 1), time=[...123.45])
+
+        """
+        if not self._need_interp():
+            # This includes the special case of a single time point (where
+            # `interp1d` would not work):
+            return Stimulus(self.data[:, 0], electrodes=self.electrodes,
+                            time=None, compress=self.compressed)
+
+        # Should work with scalars and lists:
+        time = np.array([time]).flatten()
+        data = np.array([[self._interp[e](t) for t in time]
+                         for e, _ in enumerate(self.electrodes)])
+        return Stimulus(data, electrodes=self.electrodes, time=time,
+                        compress=self.compressed)
+
+    def __eq__(self, other):
+        """Returns True if two Stimulus objects are identical
+
+        Two Stimulus objects are considered identical if they have the same
+        electrode names, time steps, and data points.
+
+        Parameters
+        ----------
+        other : any
+            Another object or variable to which the current object should be
+            compared.
+
+        Examples
+        --------
+        >>> from pulse2percept.stimuli import Stimulus
+        >>> stim1 = Stimulus(np.ones(3))
+        >>> stim2 = Stimulus(np.zeros(5))
+        >>> stim1 == stim2
+        False
+
+        Compare a Stimulus with something else entirely:
+
+        >>> from pulse2percept.stimuli import Stimulus
+        >>> stim1 = Stimulus(np.ones(3))
+        >>> stim1 == 1
+        False
+
+        """
+        if not isinstance(other, Stimulus):
+            return False
+        if self.time is None:
+            if other.time is not None:
+                return False
+        else:
+            if other.time is None:
+                return False
+            if len(self.time) != len(other.time):
```

Comment:
This means that two stimuli would be unidentical if one is a "compressed" version of the other. Meaning that you could have: 

```
>>> s1 == s2 
False
>>> s1.compress()
>>>s1 == s2 
True
```

That seems fine to me, but want to make sure that's intended.